packages/react-flow-editor/src/Editor/components/Scale/useOverview.ts

```diff
@@ -1,68 +1,16 @@
-import { useStore } from "@nanostores/react"
-import { useEffect, useState } from "react"
+import { useEffect } from "react"
 
-import { NodesAtom, TransformationMap } from "@/Editor/state"
-import { DRAG_OFFSET_TRANSFORM, LARGEST_RECT } from "@/Editor/constants"
 import { useRectsContext } from "@/Editor/rects-context"
+import { overviewActions } from "@/Editor/state"
 
 export const useOverview = () => {
-  const { zoomContainer, editorContainer } = useRectsContext()
+  const { editorContainerRef, isMounted } = useRectsContext()
 
-  const transformation = useStore(TransformationMap)
-  const [underOverview, setUnderOverview] = useState<boolean>(false)
+  const makeOverview = () => {
+    if (isMounted && editorContainerRef.current) overviewActions.overview(editorContainerRef)
+  }
 
-  useEffect(() => {
-    if (underOverview && editorContainer) {
-      const nodes = NodesAtom.get()
+  useEffect(makeOverview, [isMounted])
 
-      if (nodes.length) {
-        const editorRect = editorContainer.getBoundingClientRect()
-
-        const dimensionsRect = nodes.reduce(
-          (acc, node) => {
-            if (node.position.x > acc.rightPoint)
-              acc.rightPoint = node.position.x + (node.rectPosition?.width || 0) / transformation.zoom
-            if (node.position.x < acc.leftPoint) acc.leftPoint = node.position.x
-            if (node.position.y > acc.bottomPoint)
-              acc.bottomPoint = node.position.y + (node.rectPosition?.height || 0) / transformation.zoom
-            if (node.position.y < acc.topPoint) acc.topPoint = node.position.y
-
-            return acc
-          },
-          { ...LARGEST_RECT }
-        )
-
-        const width = dimensionsRect.rightPoint - dimensionsRect.leftPoint
-        const height = dimensionsRect.bottomPoint - dimensionsRect.topPoint
-
-        const newZoom = Math.min(
-          (editorRect.width + DRAG_OFFSET_TRANSFORM) / width,
-          (editorRect.height + DRAG_OFFSET_TRANSFORM) / height
-        )
-        const newZoomCorrected = newZoom > 1 ? 1 : newZoom
-
-        const dx = -dimensionsRect.leftPoint + (editorRect.width - width) / 2
-        const dy = -dimensionsRect.topPoint + (editorRect.height - height) / 2
-
-        zoomContainer.style.transformOrigin = `${dx}px ${dy}px`
-
-        TransformationMap.set({
-          dx,
-          dy,
-          zoom: newZoomCorrected
-        })
-      }
-
-      setUnderOverview(false)
-    }
-  }, [underOverview, transformation, editorContainer])
-
-  /** Overview on mount */
-  useEffect(() => {
-    setUnderOverview(true)
-  }, [])
-
-  const overview = () => setUnderOverview(true)
-
-  return overview
+  return makeOverview
```

Comment:
Strange return, and useOverview doesn't have a return type

Reply:
why strange, we need to make overview on mount, so we need to use it in useeefect, it was logically combined in one hook, but we need function outside